CMakeLists.txt

```diff
@@ -29,9 +29,14 @@ target_link_libraries(effil -lpthread ${LUA_LIBRARY} -ldl)
 set_target_properties(effil PROPERTIES SUFFIX .so)
 
 set(GENERAL "-std=c++14 -DSOL_EXCEPTIONS_SAFE_PROPAGATION")
-set(ENABLE_WARNINGS "-Wall -Wextra -pedantic")
+if(CMAKE_COMPILER_IS_GNUCXX)
```

Comment:
We also use clang for macos, so I think it's better to use `if(NOT WIN32)` or even `if(NOT MSVC)`

Reply:
I will go with `if(NOT MSVC)`, because one may use gcc/clang on Windows (probably in cygwin).